src/Compilers/Core/Portable/RuntimeCapability.cs

```diff
@@ -0,0 +1,88 @@
+        /// </summary>
+        CovariantReturnsOfClasses = 2,
+
+        /// <summary>
+        /// Represents a runtime feature where interfaces can define default method implementations.
+        /// </summary>
+        DefaultImplementationsOfInterfaces = 3,
+
+        /// <summary>
+        /// Indicates that this version of the runtime supports IntPtr and UIntPtr as numeric types.
+        /// </summary>
+        NumericIntPtr = 4,
+
+        /// <summary>
+        /// Represents a runtime feature where C# function pointers can be declared with an unmanaged calling convention.
+        /// </summary>
+        UnmanagedSignatureCallingConvention = 5,
+
+        /// <summary>
+        /// Indicates that this version of runtime supports virtual static members of interfaces.
+        /// </summary>
+        VirtualStaticsInInterfaces = 6,
+    }
+
+    internal static class RuntimeCapabilityHelpers
+    {
+        public static bool RuntimeSupportsCapability(IAssemblySymbolInternal assembly, RuntimeCapability capability)
+        {
+            switch (capability)
+            {
+                case RuntimeCapability.ByRefFields:
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__ByRefFields);
+
+                case RuntimeCapability.CovariantReturnsOfClasses:
+                    // check for the runtime feature indicator and the required attribute.
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__CovariantReturnsOfClasses) &&
+                           assembly.GetSpecialType(SpecialType.System_Runtime_CompilerServices_PreserveBaseOverridesAttribute) is { TypeKind: TypeKind.Class };
+
+                case RuntimeCapability.DefaultImplementationsOfInterfaces:
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__DefaultImplementationsOfInterfaces);
+
+                case RuntimeCapability.NumericIntPtr:
+                    // CorLibrary should never be null, but that invariant is broken in some cases for MissingAssemblySymbol.
+                    // Tracked by https://github.com/dotnet/roslyn/issues/61262
+                    return assembly.CorLibrary is not null &&
+                           RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__NumericIntPtr);
+
+                case RuntimeCapability.UnmanagedSignatureCallingConvention:
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__UnmanagedSignatureCallingConvention);
+
+                case RuntimeCapability.VirtualStaticsInInterfaces:
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__VirtualStaticsInInterfaces);
```

Comment:
the logic of these was pulled from the C# impl.  I'm not sure why some are different (like why NumericIntPtr explicitly checks .CorLibrary, but the rest do not).

Reply:
The reason NumericIntPtr from the others is that it is involved in metadata decoding. This flag affects how we read `System.IntPtr`. 
There's a bug tracking that it is possible to have null CorLibrary, so it needed special handling. Once that bug is fixed, the logic should be simple like the other cases.

Reply:
All the others end up calling into GetSpecialTypeMember, which does thsi:

```c#
        internal virtual Symbol GetSpecialTypeMember(SpecialMember member)
        {
            return CorLibrary.GetDeclaredSpecialTypeMember(member);
        }
```

So it looks like the others will just crash if you have a null CorLibrary. :)